charts/livy/templates/headless-service.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "livy.fullname" . }}-headless
+  labels:
+    app.kubernetes.io/name: {{ include "livy.name" . }}
+    helm.sh/chart: {{ include "livy.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  type: ClusterIP
+  clusterIP: None
+  ports:
```

Comment:
Do we need this block though? It should work without `spec.ports` at all.
And I would prefer to call this file `service-headless.yaml`

Reply:
We can add a flag or remove it altogether if you prefer. I've researched on the headless service topic and most charts that use StatefulSets have both a headless and regular service because the docs says you must: `StatefulSets currently require a Headless Service to be responsible for the network identity of the Pods. You are responsible for creating this Service.` But in the Headless Service docs, its said: `Sometimes you don’t need load-balancing and a single Service IP. In this case, you can create what are termed “headless” Services`. So it seems that if you already use a regular service it can be used as a replacement for the headless one without problems. In fact I only added the headless one few weeks ago to comply with the "standards".
I Found [this k8s issue](https://github.com/kubernetes/kubernetes/issues/69608) that says you can even put `foobar` as serviceName if you don't care about the pod been reacheable by its hostname. Its not our case here, but implies that it works with regular service, headless service and no service at al.